src/basic/efivars.c

```diff
@@ -353,9 +362,40 @@ int systemd_efi_options_variable(char **line) {
 
                 *line = m;
                 return 0;
+        } else {
+                r = read_one_line_file(EFIVAR_CACHE_PATH(EFI_VARIABLE_SystemdOptions), line);
+                if (r == -ENOENT)
+                        return -ENODATA;
+                return r;
+        }
+}
+
+static inline int compare_stat_mtime(const struct stat *a, const struct stat *b) {
+        return CMP(a->st_mtim.tv_sec, b->st_mtim.tv_sec) ?:
+               CMP(a->st_mtim.tv_nsec, b->st_mtim.tv_nsec);
+}
+
+int systemd_efi_options_efivarfs_if_newer(char **line) {
+        struct stat a = {}, b = {};
```

Comment:
why initialize?

Reply:
ENOENT is ignored and the structure would be left unitialized. I'm relying on mtime being 0 for it to compare lower.